### PR TITLE
update gloo install guide for knative serving 0.8.0

### LINKED ISSUE #1690: Update Gloo docs for Knative v0.8.0
**Describe the change you'd like to see**
Update the Knative docs to reflect new features in Gloo as well as support for Serving v0.8.0

### PR DESCRIPTION
**Pull request by ilackarms**
Fixes #1690

## Proposed Changes

- Update docs for Gloo installation & basic getting started

### CODE REVIEW

**Comment by tcnghia** [comment]
/assign @RichieEscarez 

Hi Richie, can you please help with approval?  Thanks

File: docs/install/Knative-with-Gloo.md [hunk 1]
```diff
@@ -69,196 +80,95 @@ rerun the command to see the current status.
...
-# Push the container to docker registry
-docker push {username}/helloworld-go
-```
+> Note that when following other Knative tutorials, you'll need to connect to the Gloo Gateway rather than the Istio Gateway when the tutorials prompts doing so.
 
-After the build has completed and the container is pushed to docker hub, you can
-deploy the app into your cluster. Ensure that the container image value in
-`service.yaml` matches the container you built in the previous step. Apply the
-configuration using `kubectl`:
+To get the URL of the Gloo Gateway, run
```
Comment (RichieEscarez) [inline docs/install/Knative-with-Gloo.md]: Move this above "Whats next"

```suggestion

## Connecting to the Gloo Gateway

Use the Gloo Gateway IP address and port number to connect to other services in your Knative cluster.

To get the URL of the Gloo Gateway, run
```

File: docs/install/Knative-with-Gloo.md [hunk 2]
```diff
@@ -69,196 +80,95 @@ rerun the command to see the current status.
...
-spec:
-  template:
-    spec:
-      containers:
-        - image: docker.io/{username}/helloworld-go
-          env:
-            - name: TARGET
-              value: "Go Sample v1"
-```
+## What's next
```
Comment (RichieEscarez) [inline docs/install/Knative-with-Gloo.md]: Please move this section to the bottom at the very end

File: docs/install/Knative-with-Gloo.md [hunk 3]
```diff
@@ -16,13 +22,11 @@ enabled. `kubectl` v1.10 is also required. This guide assumes that you've
 already created a Kubernetes cluster which you're comfortable installing _alpha_
 software on.
 
-This guide assumes you are using bash in a Mac or Linux environment; some
-commands will need to be adjusted for use in a Windows environment.
+This guide assumes you are using bash in a Mac or Linux environment; some commands will need to be adjusted for use in a Windows environment.
```
Comment (RichieEscarez) [inline docs/install/Knative-with-Gloo.md]: @tcnghia  @ZhiminXiang is this correct? Are there other features that should be mentioned here?
```suggestion

- Some Knative features might not include support for Gloo. For example, the [automatic TLS certificate provisioning](../serving/using-auto-tls.md) is currently only supported in Istio.
```

File: docs/install/Knative-with-Gloo.md [hunk 4]
```diff
@@ -1,12 +1,18 @@
...
+linkTitle: "Knative with Gloo"
 weight: 10
 type: "docs"
 ---
 
-This guide walks you through the installation of the latest version of Knative
-using pre-built images.
+[Gloo](https://gloo.solo.io) is a popular open-source Envoy control plane and API gateway built for Kubernetes (and other platforms). 
+
+Gloo can function as a complete replacement for Istio, supporting the full Knative Ingress spec. Using Knative with Gloo requires less resource usage and operational overhead than Istio.
```
[the remark this answers is not in the document]
Comment (ilackarms) [reply docs/install/Knative-with-Gloo.md]: technically Gloo does not provide a `service mesh` replacement for Istio but a `gateway` replacement. I'll push that change

File: docs/install/Knative-with-Gloo.md (hunk 3 above)
Comment (ilackarms) [reply docs/install/Knative-with-Gloo.md]: this is incorrect, Gloo supports the Automatic TLS feature (since it's included in the `Ingress` spec)

**Comment by ilackarms** [comment]
@RichieEscarez i confirmed that Gloo *is* working with (Automatic) TLS provisioning, so I've updated the docs to reflect this

**Comment by ilackarms** [comment]
@RichieEscarez @tcnghia i've updated this PR with the suggested changes and it should be ready for another review. much appreciated!

File: docs/install/Knative-with-Gloo.md [hunk 5]
```diff
@@ -16,13 +22,11 @@ enabled. `kubectl` v1.10 is also required. This guide assumes that you've
 already created a Kubernetes cluster which you're comfortable installing _alpha_
 software on.
 
-This guide assumes you are using bash in a Mac or Linux environment; some
-commands will need to be adjusted for use in a Windows environment.
+This guide assumes you are using bash in a Mac or Linux environment; some commands will need to be adjusted for use in a Windows environment.
```
[the remark this answers is not in the document]
Comment (RichieEscarez) [reply docs/install/Knative-with-Gloo.md]: Do you know which commands "will need to be adjusted"?

File: docs/install/Knative-with-Gloo.md [hunk 6]
```diff
@@ -16,13 +22,11 @@ enabled. `kubectl` v1.10 is also required. This guide assumes that you've
...
 
 ## Installing Glooctl
```
[the remark this answers is not in the document]
Comment (RichieEscarez) [reply docs/install/Knative-with-Gloo.md]: if not with code highlighting, this should be lower case (sentence style capitalization for section titles)

File: docs/install/README.md [hunk 7]
```diff
@@ -36,10 +36,7 @@ component.
 
 ## Installing Knative with Gloo
 
-[Install with Gloo](./Knative-with-Gloo.md): Gloo functions as a lightweight
-gateway for Knative. Choose this option if you don't require service mesh in
-your cluster and want a lighter-weight alternative to Istio. Note that _the
-Knative Eventing component_ is not supported by Gloo at this time.
+[Install with Gloo](./Knative-with-Gloo.md): Gloo functions as a lightweight gateway for Knative. Choose this option if you don't require service mesh in your cluster and want a lightweight alternative to Istio. Gloo supports all documented Knative features, as well as extensions to Serving such as Eventing and Monitoring.
```
Comment (RichieEscarez) [inline docs/install/README.md]:
```suggestion
[Install with Gloo](./Knative-with-Gloo.md): Gloo functions as a lightweight gateway for Knative. Choose this option if you don't require a service mesh in your cluster and want a lightweight alternative to Istio. Gloo supports all documented Knative features, as well as extensions to Serving such as Eventing and Monitoring.
```

File: docs/install/Knative-with-Gloo.md [hunk 8]
```diff
@@ -1,12 +1,16 @@
...
 ---
 
-This guide walks you through the installation of the latest version of Knative
-using pre-built images.
+Learn how to deploy Gloo and Knative to your Kubernetes cluster using the Gloo command line tool `glooctl`.
+ 
+
+[Gloo](https://gloo.solo.io) is a popular open-source Envoy control plane and API gateway built for Kubernetes (and other platforms). 
+
+Gloo provides a complete gateway replacement for Istio and supports the full Knative Ingress spec. Choose Gloo as a lightweight option that requires less resource usage and operational overhead compared to Istio.
```
Comment (RichieEscarez) [inline docs/install/Knative-with-Gloo.md]:
```suggestion
Gloo provides a complete gateway replacement for Istio and supports the full Knative Ingress spec. Choose Gloo if you don't require a service mesh in your cluster and want a lightweight alternative that requires less resource usage and operational overhead.
```

File: docs/install/Knative-with-Gloo.md [hunk 9]
```diff
@@ -69,196 +81,95 @@ rerun the command to see the current status.
...
-Once the sample code has been created, we'll build and deploy it
+   Gloo uses the `Host` header to route requests to the correct
+   service. You can send a request to the `helloworld-go` service with curl
+   using the `Host` and `$GATEWAY_URL` from above:
 
-Use Docker to build the sample code into a container. To build and push with
-Docker Hub, run these commands replacing `{username}` with your Docker Hub
-username:
+   ```
+   $ curl -H "Host: helloworld-go.default.example.com" $(glooctl proxy url --name knative-external-proxy)
```
Comment (RichieEscarez) [inline docs/install/Knative-with-Gloo.md]:
```suggestion
   curl -H "Host: helloworld-go.default.example.com" $(glooctl proxy url --name knative-external-proxy)
```

**Comment by RichieEscarez** [comment]
A few more small things in addition to the several organizational comments from first round. Let me know if you feel strongly about keep the text the way it is. Also, if you prefer that I commit all the edits myself, im open to that too. Thanks again, @ilackarms

**Pull request merged**